src/game/NPCHandler.cpp

```diff
@@ -575,8 +579,18 @@ void WorldSession::HandleStablePet(WorldPacket& recv_data)
 
     Pet* pet = _player->GetPet();
 
+    if (!pet) // if pet not by your side, load from DB for a moment
```

Comment:
It's not just 'loaded for a moment', not unless you unsummon it again because LoadPetFromDB spawns it.